src/app/script/canvas_widget.cpp

```diff
@@ -150,14 +195,19 @@ void Canvas::onResize(ui::ResizeEvent& ev)
 {
   Widget::onResize(ev);
   if (os::instance() && !ev.bounds().isEmpty()) {
-    const int w = ev.bounds().w;
-    const int h = ev.bounds().h;
+    int w = ev.bounds().w;
+    int h = ev.bounds().h;
 
     if (!m_surface ||
         m_surface->width() != w ||
         m_surface->height() != h) {
-      m_surface = os::instance()->makeSurface(w, h);
 
+      if (m_autoScaling) {
+        w = w/ui::guiscale() + (w % ui::guiscale());
+        h = h/ui::guiscale() + (h % ui::guiscale());
+      }
```

Comment:
The w/h should be modified before the if() condition.

Reply:
Good catch! Will fix this. Thanks!